pymongo/periodic_executor.py

```diff
@@ -91,7 +91,14 @@ def open(self) -> None:
             thread.daemon = True
             self._thread = weakref.proxy(thread)
             _register_executor(self)
-            thread.start()
+            try:
+                thread.start()
+            except RuntimeError as e:
+                if str(e) == "can't create new thread at interpreter shutdown":
```

Comment:
We're assuming this error message will remain constant,  right?

Reply:
Yep